src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTests.cs

```diff
@@ -69,6 +69,85 @@ protected override void BuildRenderTree(global::Microsoft.AspNetCore.Components.
             Assert.Single(result.GeneratedSources);
         }
 
+        [Fact]
+        public async Task SourceGenerator_RazorFiles_DesignTime()
+        {
+            // Arrange
+            var project = CreateTestProject(new()
+            {
+                ["Pages/Index.razor"] = "<h1>Hello world</h1>",
+            });
+
+            var compilation = await project.GetCompilationAsync();
+            var (driver, additionalTexts, optionsProvider) = await GetDriverWithAdditionalTextAndProviderAsync(project, hostOutputs: true);
+
+            // Enable design-time.
+            var options = optionsProvider.Clone();
+            options.TestGlobalOptions["build_property.RazorDesignTime"] = "true";
+            driver = driver.WithUpdatedAnalyzerConfigOptions(options);
+
+            var result = RunGenerator(compilation!, ref driver);
+
+            result.VerifyPageOutput(
+@"#pragma checksum ""Pages/Index.razor"" ""{ff1816ec-aa5e-4d10-87f7-6f4963833460}"" ""6b5db227a6aa2228c777b0771108b184b1fc5df3""
```

Comment:
Nit: `"""` strings would be more readable.

Reply:
Yes, but the test outputs the `@"` string as a hint if you don't provide one when you first write the test, so I guess we should keep this consistent with the string you would copy-paste from the test output:

https://github.com/dotnet/razor/blob/d26d2072b8f935b39801857514bafd5bdf2bb48c/src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTestsBase.cs#L319

Reply:
We can go update that helper to use `"""` strings in a separate PR :).